3.6/debian-9/rootfs/opt/bitnami/scripts/libmongodb.sh

```diff
@@ -13,6 +13,85 @@
 . /opt/bitnami/scripts/libfs.sh
 . /opt/bitnami/scripts/libnet.sh
 
+########################
+# Create alias for environment variable, so both can be used
+# Globals:
+#   None
+# Arguments:
+#   $1 - Alias environment variable name
+#   $2 - Original environment variable name
+# Returns:
+#   None
+#########################
+mongodb_declare_alias_env() {
+    local -r alias="${1:?missing environment variable alias}"
+    local -r original="${2:?missing original environment variable}"
+    if printenv "${original}" > /dev/null; then
+        export "$alias"="${!original:-}"
+    fi
+}
+
+
+########################
+# Map MongoDB legacy environment variables to the new names
+# Globals:
+#   MONGODB_*
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
+mongodb_create_alias_environment_variables() {
+    mongodb_declare_alias_env "MONGODB_USERNAMES" "MONGODB_USERNAME"
+    mongodb_declare_alias_env "MONGODB_PASSWORDS" "MONGODB_PASSWORD"
+    mongodb_declare_alias_env "MONGODB_DATABASES" "MONGODB_DATABASE"
+    mongodb_declare_alias_env "ALLOW_EMPTY_PASSWORDS" "ALLOW_EMPTY_PASSWORD"
+}
+
+
+########################
+# Return field separator to use in lists. One of comma or semi-colon, comma
+# being preferred.
+# Globals:
+#   None
+# Arguments:
+#   A (list) of fields
+# Returns:
+#   The separator used within that list
+#########################
+mongodb_field_separator() {
+    if printf %s\\n "$1" | grep -q ','; then
+        printf %s\\n ','
+    elif printf %s\\n "$1" | grep -q ';'; then
+        printf %s\\n ';'
+    fi
+}
+
+
+########################
+# Initialise the arrays databases, usernames and passwords to contain the
+# fields from their respective environment variables. 
+# Globals:
+#   MONGODB_DATABASES, MONGODB_USERNAMES, MONGODB_PASSWORDS
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
+mongodb_auth() {
+    IFS="$(mongodb_field_separator "${MONGODB_DATABASES}")" read -r -a databases <<< "${MONGODB_DATABASES}"
+    IFS="$(mongodb_field_separator "${MONGODB_USERNAMES:-}")" read -r -a usernames <<< "${MONGODB_USERNAMES:-}"
+    IFS="$(mongodb_field_separator "${MONGODB_PASSWORDS:-}")" read -r -a passwords <<< "${MONGODB_PASSWORDS:-}"
```

Comment:
Shouldn't this be enough?
```suggestion
read -r -a databases <<< "$(tr ',;' ' ' <<< "${MONGODB_DATABASES}")"
read -r -a users <<< "$(tr ',;' ' ' <<< "${MONGODB_USERNAMES}")"
read -r -a passwords <<< "$(tr ',;' ' ' <<< "${MONGODB_PASSWORDS}")"
```

Reply:
No, I don't think so.
+ I believe that your implementation would prevent the use of space, tabs and newlines in the passwords, as these are the regular delimiters in shell. (of those maybe space is the one to look for).
+ Also the use of `${xx:-}` was conscious to permit reading from empty and non-existing variables.

Reply:
The `${xx:-}` shouldn't be necessary if we declare the env. vars at `mongodb-env.sh`. Regarding the allowed delimiters you're right, my suggestion will treat spaces, tabs, newlines, commas and semicolons as delimiters.
My question is: does MongoDB accepts spaces, tabs and newlines in the passwords?

Reply:
Yes. MongoDB accepts spaces in the passwords. The documentation does not [mention](https://docs.mongodb.com/manual/reference/method/db.updateUser/) any restrictions for passwords and I verified through creating a Mongo instance with a root password that contained a space. I would suggest that we keep my version of the code.

Reply:
All right! Let's use whatever approach that is less restrictive with what characters can be included in the passwords.